### DISCUSSION TITLE
Unexpected problem installing iqtree-2.3.2-macOS-arm

### DISCUSSION THREAD

**Discussion by lsjermiin** [iqtree/iqtree2#192] [Q&A]
Hi,
I just downloaded IQTREE2 v2.3.2-macOS-arm and installed it on my MacBook Pro (Apple M2 Pro; MacOS 13.5.1). I then tested the program using the 

iqtree2 --help

command, and got the following error message:

dyld[2516]: Library not loaded: /opt/homebrew/opt/libomp/lib/libomp.dylib
  Referenced from: <09FB5B21-1D4E-3780-926D-6607370AB034> /usr/local/bin/iqtree2
  Reason: tried: '/opt/homebrew/opt/libomp/lib/libomp.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libomp/lib/libomp.dylib' (no such file), '/opt/homebrew/opt/libomp/lib/libomp.dylib' (no such file)
zsh: abort      iqtree2 -h

This is the first time in the last 10 years I have used iqtree that I have encountered this problem.

What is the shorter fix? Go to homebrew and download the libomp.dylib library?

What is the longterm fix? I ask this because I use iqtree in my teaching of 3rd-year students, and part of the education involves them downloading software and installing it on their computer (Windows, Mac, Linux, etc). Having to install a library as well will be a deterrent for most novices in Bioinformatics, especially those who are not from IT (e.g., biologists).

Thanks,

Lars

**Comment by thomaskf** [comment]
Hi Lars,

Thanks for the question.

libomp is an openmp library used for multithreading programs. IQ-TREE
requires this library because it uses multithreading to speed up the
computation.

Please install this library by using the following command:
> brew install libomp

I guess you have this library installed on your previous MacBook, and thus,
you didn't have this problem before.

Best regards,
Thomas

**Comment by lsjermiin** [comment]
Dear Thomas,

Two things:

  1.  The command you suggested is not working: zsh: command not found: brew
  2.  I have had many previous versions of iqtree2 working on this computer. This is a new problem.
Still trying to fix the problem.
All the best,
Lars

From: Thomas Wong ***@***.***>
Date: Thursday, 9 May 2024 at 23:07
To: iqtree/iqtree2 ***@***.***>
Cc: Jermiin, Lars ***@***.***>, Author ***@***.***>
Subject: Re: [iqtree/iqtree2] Unexpected problem installing iqtree-2.3.2-macOS-arm (Discussion #192)
EXTERNAL EMAIL: This email originated outside the University of Galway. Do not open attachments or click on links unless you believe the content is safe.
RÍOMHPHOST SEACHTRACH: Níor tháinig an ríomhphost seo ó Ollscoil na Gaillimhe. Ná hoscail ceangaltáin agus ná cliceáil ar naisc mura gcreideann tú go bhfuil an t-ábhar sábháilte.

Hi Lars,

Thanks for the question.

libomp is an openmp library used for multithreading programs. IQ-TREE
requires this library because it uses multithreading to speed up the
computation.

Please install this library by using the following command:
> brew install libomp

I guess you have this library installed on your previous MacBook, and thus,
you didn't have this problem before.

Best regards,
Thomas

**Comment by thomaskf** [comment]
Dear Lars,
The issue has been resolved. We have updated the binary files for the Mac ARM processor. Please download the file and try again.
Thank you very much for reporting this problem. Appreciated! We have identified two issues, and they have both been resolved.
Best regards,
Thomas

**Comment by lsjermiin** [comment]
Thanks for fixing that bug -- much appreciated!!
All the best,
Lars

**Comment by lsjermiin** [comment]
Dear Thomas,
Thanks a lot for that – it works. I have closed the case on GitHub.
All the best,
Lars

From: Thomas Wong ***@***.***>
Date: Saturday, 11 May 2024 at 02:43
To: iqtree/iqtree2 ***@***.***>
Cc: Jermiin, Lars ***@***.***>, Author ***@***.***>
Subject: Re: [iqtree/iqtree2] Unexpected problem installing iqtree-2.3.2-macOS-arm (Discussion #192)
EXTERNAL EMAIL: This email originated outside the University of Galway. Do not open attachments or click on links unless you believe the content is safe.
RÍOMHPHOST SEACHTRACH: Níor tháinig an ríomhphost seo ó Ollscoil na Gaillimhe. Ná hoscail ceangaltáin agus ná cliceáil ar naisc mura gcreideann tú go bhfuil an t-ábhar sábháilte.


Dear Lars,
The issue has been resolved. We have updated the binary files for the Mac ARM processor. Please download the file and try again.
Thank you very much for reporting this problem. Appreciated! We have identified two issues, and they have both been resolved.
Best regards,
Thomas
